runtime/src/iree/hal/drivers/hip/hip_device.c

```diff
@@ -77,7 +77,7 @@ typedef struct iree_hal_hip_device_t {
   // are met. It buffers submissions and allocations internally before they
   // are ready. This queue couples with HAL semaphores backed by iree_event_t
   // and hipEvent_t objects.
-  iree_hal_hip_pending_queue_actions_t* pending_queue_actions;
```

Comment:
this is how it was and not something done in this PR, but something to think about is how we'll extend this for multiple queues (hopefully just making this an array and selecting the right one on the queue operations) - the nice thing about all the logic being wrapped up in the work queue is that we can move anything queue specific that's bled out (hip_dispatch_stream, tracing_context, etc) into the work queue and stamp them out as multiple per physical device or one or more per physical device.